### ISSUE TITLE
Sharing: Add label for honeypot

### ISSUE THREAD

**Issue by kraftbj** [Automattic/jetpack#3260]
The Share by e-mail form includes a spam honeypot, but we do not include a label or other hint for screen readers to indicate the use of the field.

We should add a label with something along the lines of "This field is for validation and should be left unchanged."

Reported in 2484787-t

**Comment by kraftbj** [comment]
Report from accessibility testing tool: 
![screen shot 2016-01-13 at 10 05 47](https://cloud.githubusercontent.com/assets/88897/12299661/5c2e1410-b9dd-11e5-924e-39702efe2851.png)

**Comment by jeherve** [comment]
Related: #448
